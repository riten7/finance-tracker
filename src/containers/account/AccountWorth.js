import React from 'react';
import { useSelector } from 'react-redux';
import { Typography } from 'antd';

const { Text } = Typography;

const AccountWorth = () => {
  const { accounts } = useSelector(state => state.accounts);
  const totalBalance = accounts.reduce((acc, account) => {
    return acc + parseInt(account.balance)
  }, 0);
  return (
    <div>
    <h2 className="account-total">Total: <Text type={totalBalance > 0 ? 'success' : 'danger'} code><strong>{`NPR. ${totalBalance}`}</strong></Text></h2>
    <br />
    </div>
  );
}

export default AccountWorth;