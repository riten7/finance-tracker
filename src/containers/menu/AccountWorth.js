import React from 'react';
import { useSelector } from 'react-redux';
import { Typography } from 'antd';

const { Text } = Typography;

const AccountWorth = () => {
  const { accounts } = useSelector(state => state.accounts);
  const getTotalWorth = () => accounts.reduce((acc, account) => {
    return  acc + parseInt(account.balance);
  }, 0)
  return (
    <div>
    <h2 className="account-total">Total: <Text code><strong>NPR. {getTotalWorth()}</strong></Text></h2>
    <br />
    </div>
  );
}

export default AccountWorth;