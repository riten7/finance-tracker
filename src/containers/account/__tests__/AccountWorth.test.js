import React from 'react';
import { shallow } from "enzyme";
import { Provider } from 'react-redux';
import configureStore from "redux-mock-store";
import AccountWorth from '../AccountWorth';
import data from '../../../__mockData__/data.json';

describe("AccountWorth component", () => {
  let wrapper;
  const store = configureStore()(data);
   it("Component renders properly", () => {
    wrapper = shallow(<Provider store={store}><AccountWorth /></Provider>);
    expect(wrapper.html()).toMatchSnapshot();
  });
});