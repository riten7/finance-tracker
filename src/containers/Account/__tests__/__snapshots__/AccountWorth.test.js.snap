// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component Component renders properly 1`] = `"<div><h2 class=\\"account-total\\">Total: <span class=\\"ant-typography\\"><code><strong>NPR. 131406817</strong></code></span></h2><br/></div>"`;
